Environment shells (#1248)
orchest/orchest: docs/source/fundamentals/environments.md

🚨 Do not install packages by running {code}`!pip install <package-name>` inside your Jupyter Notebook. This causes the package to be installed every time you run the Pipeline step. It is not saved in the environment as containers are stateless!
```

## Environment shells

> JupyterLab has integrated support for shell sessions using a terminal emulator. To run your scripts and CLI commands while developing your data pipeline you can make use of shells that are based on the environments you have in your project. The environment container image is used such that you have access to all installed programs/dependencies.

Reviewer: We choose to capitalize all terms in Orchest, so `environment` should be `Environment` ([our implementation detail docs](https://docs.orchest.io/en/stable/development/implementation_best_practices.html#writing-docs)). 

```suggestion
JupyterLab has integrated support for shell sessions using a terminal emulator. To run your scripts and CLI commands while developing your data Pipeline you can make use of shells that are based on the Environments you have in your Project. The Environment container image is used such that you have access to all installed programs/dependencies.
```

> All you need to do is open JupyterLab in a session and click on Terminal:

Reviewer: ```suggestion
All you need to do is open JupyterLab and click on Terminal:
```

Don't think `in a session` adds anything here (especially given that we start the Session automatically).

```{image} ../img/environment-shells-jupyterlab.png
:width: 800
:alt: Terminal in JupyterLab
:align: center
```

```{warning}

> If you make any changes to your environment, you'll need to restart your environment shell to see the changes. You can restart your environment shell by exiting out of it by typing Ctrl + D or typing 'exit'.

Reviewer: ```suggestion
If you make any changes to your environment, you'll need to restart your environment shell to see the changes. You can restart your environment shell by exiting out of it by pressing 'Ctrl + D' or typing 'exit'.
```
